### PR TITLE
NCSDK-30461: Move SUIT MPI to sysbuild

### PR DESCRIPTION
**Pull request by tomchy**

### CODE REVIEW

**Review by nordicjm — APPROVED** [review]

File: cmake/sysbuild/suit.cmake [hunk 1]
```diff
@@ -528,19 +527,11 @@ function(suit_setup_merge)
...
-    if(${IMAGE_TARGET_NAME} STREQUAL "application")
-      # Get a list of files (and their dependencies) which need merging into the uicr merged file
-      # and add them at the end of the list, allowing for overwriting, this only applies to the
-      # application core image
-      get_property(merge_files GLOBAL PROPERTY SUIT_MERGE_FILE)
-      get_property(merge_dependencies GLOBAL PROPERTY SUIT_MERGE_DEPENDENCIES)
-    else()
+    # Get a list of files (and their dependencies) which need merging into the uicr merged file
+    # and add them at the end of the list, allowing for overwriting
+    get_property(merge_files GLOBAL PROPERTY SUIT_MERGE_${IMAGE_TARGET_NAME}_FILE)
```
Comment (ahasztag) [inline cmake/sysbuild/suit.cmake:532]: I think `unset(merge_files)` is needed for the if clause below to work properly - or at least it is a bit cleaner

File: config/suit/templates/nrf54h20/default/v1/app_recovery_envelope_direct.yaml.jinja2 [hunk 2]
```diff
@@ -1,10 +1,10 @@
 {%- set component_index = 0 %}
 {%- set component_list = [] %}
 {%- set dependencies_list = [] %}
-{%- set mpi_app_recovery_vendor_name = application['config']['CONFIG_SUIT_MPI_APP_RECOVERY_VENDOR_NAME']|default('nordicsemi.com') %}
```
Comment (ahasztag) [inline config/suit/templates/nrf54h20/default/v1/app_recovery_envelope_direct.yaml.jinja2:4]: It seems similar changes are missing from `app_recovery_local_envelope.yaml.jinja2` and `app_recovery_local_envelope.yaml.jinja2`

File: sysbuild/CMakeLists.txt [hunk 3]
```diff
@@ -557,15 +561,6 @@ function(${SYSBUILD_CURRENT_MODULE_NAME}_pre_cmake)
     endif()
   endif()
 
-  if(SB_CONFIG_SUIT_BUILD_RECOVERY)
```
Comment (ahasztag) [inline sysbuild/CMakeLists.txt:560]: After deleting this part appropriate `select` statements should be added in the `Kconfig.suit` file

**Review by ahasztag — CHANGES_REQUESTED** [review]

**Review by robertstypa — APPROVED** [review]

File: sysbuild/CMakeLists.txt (hunk 3 above)
Comment (tomchy) [reply sysbuild/CMakeLists.txt:560]: I think changes in `sysbuild/suit_provisioning/Kconfig.nrf54h20` should be sufficient.. aren't they?

**Review by doki-nordic — APPROVED** [review]

**Review by ahasztag — CHANGES_REQUESTED** [review]
When I am attempting to flash `sample.suit.smp_transfer.recovery` without any modifications I get the following failure

`Error: One or more batch tasks failed:
 - [Probe] Failed to parse Hex file: Hex file error: No such file or directory (os error 2), code: Generic`

**Comment by tomchy** [comment]
> When I am attempting to flash `sample.suit.smp_transfer.recovery` without any modifications I get the following failure
> 
> `Error: One or more batch tasks failed:
> 
>     * [Probe] Failed to parse Hex file: Hex file error: No such file or directory (os error 2), code: Generic`

Fixed in sdk-zephyr. Please run `west update`.

**Review by ahasztag — APPROVED** [review]
OK, now everything works fine from the recovery perspective

File: doc/nrf/app_dev/config_and_build/sysbuild/sysbuild_forced_options.rst [hunk 4]
```diff
@@ -65,10 +65,6 @@ Trying to change them directly in an image will result in the new value being ov
...
 | :kconfig:option:`CONFIG_CHIP_OTA_IMAGE_BUILD`                           +                                                                           |                         |
 +-------------------------------------------------------------------------+---------------------------------------------------------------------------+                         |
-| :kconfig:option:`CONFIG_SUIT_MPI_APP_RECOVERY`                          +               ``SB_CONFIG_SUIT_BUILD_RECOVERY``                           |                         |
-+-------------------------------------------------------------------------+                                                                           |                         |
-| :kconfig:option:`CONFIG_SUIT_MPI_RAD_RECOVERY`                          +                                                                           |                         |
-+-------------------------------------------------------------------------+---------------------------------------------------------------------------+-------------------------+
```
Comment (b-gent) [inline doc/nrf/app_dev/config_and_build/sysbuild/sysbuild_forced_options.rst:71]: This change results in the merge of the two rows in the last column. You can see the result here -> https://ncsdoc.z6.web.core.windows.net/PR-19016/nrf/app_dev/config_and_build/sysbuild/sysbuild_forced_options.html

For comparison, this is the current output of this file before your change -> https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/app_dev/config_and_build/sysbuild/sysbuild_forced_options.html

Is this change intended?

**Review by ArekBalysNordic — APPROVED** [review]
Checked with Matter. Changes in envelope templates look OK.

File: doc/nrf/app_dev/config_and_build/sysbuild/sysbuild_forced_options.rst (hunk 4 above)
Comment (tomchy) [reply doc/nrf/app_dev/config_and_build/sysbuild/sysbuild_forced_options.rst:71]: No, thank you for spotting this!
Fixed.

**Review by b-gent — APPROVED** [review]
doc looks good

**Review by adsz-nordic — APPROVED** [review]

**Pull request merged**